helper/helper_test.go

```diff
@@ -98,4 +280,239 @@ var _ = Describe("helper tests", func() {
 			Expect(isSubset).To(Equal(false))
 		})
 	})
+
+	Describe("doRestoreAgent Mocked unit tests", func() {
+		BeforeEach(func() {
+			// Setup mocked tests environment
+			*singleDataFile = false
+			*content = 1
+			*oidFile = "testoid.dat"
+			*isResizeRestore = true
+			*origSize = 5
+			*destSize = 3
+			*pipeFile = "mock"
+			*onErrorContinue = true
+		})
+		It("successfully restores using a single data file when inputs are valid and no errors occur", func() {
+			*singleDataFile = true
+
+			oidBatch := []oidWithBatch{{oid: 1, batch: 1}}
+			steps := []helperTestStep{
+				{restorePipeWriterArgExpect: "mock_1_1", restorePipeWriterResult: true, skipFileArgTableOid: 1, skipFileResult: false},
+			}
+
+			mockHelper := newHelperTest(oidBatch, steps)
+
+			// Prepare and write the toc file
+			testDir := "" // Use local directory for the TOC file instead of default onr
+			*tocFile = fmt.Sprintf("%stest_toc.yaml", testDir)
+			writeTestTOC(*tocFile)
+			defer func() {
+				_ = os.Remove(*tocFile)
+			}()
+
+			*dataFile = "test_data.dat"
+			// Call the function under test
+			err := doRestoreAgentInternal(mockHelper)
+
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("successfully restores using multiple data files when inputs are valid and no errors occur", func() {
+			// Call the function under test
+			oidBatch := []oidWithBatch{{oid: 1, batch: 1}}
+			steps := []helperTestStep{
```

Comment:
Please make the style for steps description consistent across the file. I think it is better to do like it is done in many tests cases below, for ex:
```
expectedScenario := []helperTestStep{
				{"mock_100_0", true, -1, false, "Can open pipe for table 100, check_skip_file shall not be called"},
				{"mock_200_0", true, -1, false, "Can open pipe for table 200, check_skip_file shall not be called"},
				{"mock_200_1", false, 200, true, "Can not open pipe for table 200, check_skip_file shall called, skip file exists"},
				{"mock_200_2", true, -1, false, "Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called"},
			}
```